Dropped PHP 7.3, upgraded to `laminas/laminas-coding-standard:~2.3.0`, improved internal type documentation

Ocramius opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

Fixes #33

renovate[bot] committed:
Update dependency laminas/laminas-coding-standard to v2

Signed-off-by: Renovate Bot <bot@renovateapp.com>

| datasource | package                         | from  | to    |
| ---------- | ------------------------------- | ----- | ----- |
| packagist  | laminas/laminas-coding-standard | 1.0.0 | 2.3.0 |

Ocramius committed:
Allow installation of CS types via composer plugins

Ocramius committed:
Remove trailing whitespace from `README.md`

Ocramius committed:
Upgraded to `laminas/laminas-coding-standard:~2.3.0`

heiglandreas commented on src/Collection.php:
```diff
@@ -73,15 +71,15 @@ public function toArray()
             return $this->current();
         }
-        return;
+        return null;
```
We could move this `return null` up as well into the unhappy path.

Ocramius replied:
66b993ccdfd8b7856dcda1a9303c43627f6b0e28

heiglandreas commented on src/Collection.php:
```diff
@@ -123,13 +121,13 @@ public function current()
             return $this->cache[$this->current];
         }
-        return;
+        return null;
```
same here

Ocramius replied:
66b993ccdfd8b7856dcda1a9303c43627f6b0e28

Ocramius commented:
The last exception import broke it, sorry - will revert

heiglandreas commented on src/Node.php:
```diff
@@ -388,7 +396,8 @@ public function willBeMoved()
         } elseif ($this->newDn !== null) {
-            return ($this->dn != $this->newDn);
+            // non-strict comparison necessary here, as we compare two objects by value
+            return $this->dn != $this->newDn; // phpcs:ignore
```
Perhaps we can create an `equals` method on the DN-Object to make that more clear.

Ocramius replied:
I'd say that we leave it as-is to avoid magic changes/issues, for now, but we think of refactoring out `// phpcs:ignore` scenarios in future.

heiglandreas commented on src/Node.php:
```diff
@@ -812,11 +819,16 @@ protected function assertChangeableAttribute($name)
-        } elseif (array_key_exists($name, $rdn)) {
+        }
+
+        if (array_key_exists($name, $rdn)) {
```
:+1:

heiglandreas reviewed (approved):
Looks good.

Some ideas that I got but nothing that should be added to this PR. Will open another PR regarding those ideas.

Ocramius committed:
Simplified `Collection#current()` and `Collection#getFirst()`

The logic has been rewritten to have failure scenarios fan-out,
and the success scenario be at the end of each method.

Ref: https://github.com/laminas/laminas-ldap/pull/34#discussion_r933112620
Ref: https://github.com/laminas/laminas-ldap/pull/34#discussion_r933113001

Ocramius commented:
Thanks @heiglandreas!
